### PR TITLE
ci: Publish packages with provenance

### PR DESCRIPTION
**Pull request by amareshsm**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?
Publish packages with provenance
#### What changes did you make? (Give an overview)
 Added the --provenance flag to npm publish commands.
#### Related Issues
https://github.com/eslint/rewrite/issues/76

### CODE REVIEW

File: .github/workflows/release-please.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
             #-----------------------------------------------------------------------------
 
             - name: Publish @eslint/compat package to npm
-              run: npm publish -w packages/compat
+              run: npm publish -w --provenance packages/compat
```
[the remark this answers is not in the document]
Comment (nzakas) [reply .github/workflows/release-please.yml:57]: That's correct. `-w` indicates the workspace (directory) to publish from.

[the remark this answers is not in the document]
Comment (amareshsm) [reply .github/workflows/release-please.yml]: done.

**Review by harish-sethuraman — APPROVED** [review]
LGTM. Will leave for others to review.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Leaving open for @nzakas to verify.

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**